src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,387 @@
+            .getDescription(accountsType.getFilingDescriptionKey(), parameters);
+    }
+
+    /**
+     * Validates the ixbrl against TNEP. This validation is driven by the env. variable and it can
+     * be disable.
+     */
+    private boolean isValidIXBL() {
+
+        boolean isIxbrlValid = true;
+        if ("true".equals(getMandatoryEnvVariable(DISABLE_IXBRL_VALIDATION_ENV_VAR))) {
+            //TODO TNEP validation needs to be added. Copy logic from abridged. Next PR
+        }
+
+        return isIxbrlValid;
+    }
+
+    /**
+     * Generates the ixbrl by calling the document render service.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param requestBody - the http request request (json format)
+     * @return The location where the service has stored the generated ixbrl.
+     */
+    private String callIxbrlGenerator(AccountsType accountsType, String requestBody)
+        throws IOException {
+        if (requestBody != null) {
+            DocumentGeneratorConnection connection = getDocumentGeneratorConnection(accountsType,
+                requestBody);
+            return ixbrlGenerator.generateIXBRL(connection);
+        }
+
+        return null;
+    }
+
+    /**
+     * Create DocumentGeneratorConnectionImpl instance containing the document render's request
+     * settings. e.g. Method, Service URL, Body, etc.
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @param requestBody - the http request request (json format).
+     * @return {@link DocumentGeneratorConnection} with the document render service settings.
+     */
+    private DocumentGeneratorConnection getDocumentGeneratorConnection(
+        AccountsType accountsType,
+        String requestBody) {
+        DocumentGeneratorConnection connection = new DocumentGeneratorConnection();
+
+        connection.setRequestMethod("POST");
+        connection.setServiceURL(getServiceURL());
+        connection.setRequestBody(requestBody);
+        connection.setAuthorizationProperty(getAPIAuthorization());
+        connection.setAssetId(accountsType.getAssetId());
+        connection.setContentType(MediaType.TEXT_HTML_VALUE);
+        connection.setAcceptType(MediaType.TEXT_HTML_VALUE);
+        connection.setLocation(getIXBRLLocation(accountsType));
+        connection.setTemplateName(accountsType.getTemplateName());
+        connection.setSetDoOutPut(true);
+
+        return connection;
+    }
+
+    /**
+     * get the location the document render service needs to stored the ixbrl document. e.g.
+     * "s3://dev-pdf-bucket/chs-dev/accounts/small_full_accounts"
+     *
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @return
+     */
+    private String getIXBRLLocation(AccountsType accountsType) {
+        if (isSmallFullType(accountsType.getAccountType())) {
```

Comment:
All relevant information about the account type is contained within the enum, and at this point you already have the account type so providing a further check to see if its a small-full doesnt seem necessary.

Reply:
I added it since as I am thinking to future proof it; depending on the Account Type you passed in the ixbrl could be different. e.g. one for abridged, one for account, etc. 

but this is something tghta this could be added in the future  once **abridged** is migrated.

Reply:
Changes for pr's comments: dffd958